nebula_ros/schema/Pandar128E4X.schema.json

```
@@ -0,0 +1,132 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "LiDAR Hesai Pandar128E4X parameters.",
+  "type": "object",
+  "definitions": {
+    "Pandar128E4X": {
+      "type": "object",
+      "properties": {
+        "host_ip": {
+          "$ref": "sub/communication.json#/definitions/host_ip"
+        },
+        "sensor_ip": {
+          "$ref": "sub/communication.json#/definitions/sensor_ip"
+        },
+        "data_port": {
+          "$ref": "sub/communication.json#/definitions/data_port"
+        },
+        "gnss_port": {
+          "$ref": "sub/lidar_hesai.json#/definitions/gnss_port"
+        },
+        "packet_mtu_size": {
+          "$ref": "sub/communication.json#/definitions/packet_mtu_size"
+        },
+        "launch_hw": {
+          "$ref": "sub/hardware.json#/definitions/launch_hw"
+        },
+        "setup_sensor": {
+          "$ref": "sub/hardware.json#/definitions/setup_sensor"
+        },
+        "frame_id": {
+          "$ref": "sub/topic.json#/definitions/frame_id"
+        },
+        "diag_span": {
+          "$ref": "sub/topic.json#/definitions/diag_span"
+        },
+        "min_range": {
+          "$ref": "sub/misc.json#/definitions/min_range"
+        },
+        "max_range": {
+          "$ref": "sub/misc.json#/definitions/max_range"
+        },
+        "cloud_min_angle": {
+          "$ref": "sub/misc.json#/definitions/cloud_min_angle"
+        },
+        "cloud_max_angle": {
+          "$ref": "sub/misc.json#/definitions/cloud_max_angle"
+        },
+        "scan_phase": {
+          "$ref": "sub/misc.json#/definitions/scan_phase"
+        },
+        "sensor_model": {
+          "$ref": "sub/lidar_hesai.json#/definitions/sensor_model"
+        },
+        "calibration_file": {
+          "$ref": "sub/lidar_hesai.json#/definitions/calibration_file"
+        },
+        "rotation_speed": {
+          "$ref": "sub/lidar_hesai.json#/definitions/rotation_speed"
+        },
+        "return_mode": {
+          "$ref": "sub/misc.json#/definitions/return_mode",
+          "enum": [
+            "Last",
+            "Strongest",
+            "LastStrongest",
+            "First",
+            "LastFirst",
+            "FirstStrongest",
+            "Dual"
+          ]
+        },
+        "ptp_profile": {
+          "$ref": "sub/communication.json#/definitions/ptp_profile"
+        },
+        "ptp_domain": {
+          "$ref": "sub/communication.json#/definitions/ptp_domain"
+        },
+        "ptp_transport_type": {
+          "$ref": "sub/communication.json#/definitions/ptp_transport_type"
```

Only allow `L2` transport for OT128 (see datasheet).
```suggestion
          "$ref": "sub/communication.json#/definitions/ptp_transport_type",
          "enum": [
            "L2"
          ]
```